counsel.el

```diff
@@ -6379,7 +6385,44 @@ sub-directories that builds may be invoked in."
   (let ((srcdir (counsel--compile-root)))
     (when (directory-files (or blddir srcdir) nil
                            counsel-compile-make-pattern t)
-      (counsel--compile-get-make-targets srcdir blddir))))
+      (counsel--compile-get-make-targets
+       'counsel-compile--probe-make-targets srcdir blddir))))
+
+
+(defun counsel-compile--probe-make-help (dir)
+  "Return a list of Make targets based on help for DIR.
+
+It is quite common for a 'make help' invocation to return a human
+readable list of targets.  Often common targets are marked with a
+leading asterisk.  The exact search pattern is controlled by
+`counsel-compile-help-pattern'."
+  (let ((default-directory dir)
+        (primary-targets nil)
+        (targets nil))
+    ;; only proceed if the help target exists
+    (when (= (call-process "make" nil t nil "-q" "help") 1)
+      (with-temp-buffer
+        (when (= (call-process "make" nil t nil "help") 0)
+          (goto-char (point-min))
+          (while (re-search-forward counsel-compile-help-pattern nil t)
+            (let ((text (match-string-no-properties 2)))
+            (if (match-beginning 1)
+                (setq primary-targets (nconc primary-targets (list text)))
+              (setq targets (nconc targets (list text))))))))
+      (append (sort primary-targets #'string-lessp)
+              (sort targets #'string-lessp)))))
+
+(defun counsel-compile-get-make-help-invocations (&optional blddir)
+  "Have a look in the root directory for any build control files and
+query any help output.
```

Comment:
Please make the docstring's first sentence fit on a single line, as per [`(info "(elisp) Documentation Tips")`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Documentation-Tips.html).